section3/reflection.md

```diff
@@ -2,16 +2,89 @@
+
+One thing, I'm still working on improving is setting SMART goals. I still overestimate how much energy and focus I have to work. Sometimes, I won't split my work up in a manageable way and end up stressed and overwhelmed closer to the deadline of projects/work. The Marinara Pomodoro extension on Google Chrome has been helping me pace my scheduled study hours though!  
+
 1. What is a Hash, and how is it different from an Array?
 
+A hash is a collection of key:value pairs. Unlike an array, it is unordered. You can't access data in a hash using a numerical index.
+
 1. In the space below, create a Hash stored to a variable named `pet_store`.  This hash should hold an inventory of items and the number of that item that you might find at a pet store.
 
+```ruby
+pet_store = {dog_harness: 12, chew_toy: 13, hamster_ball: 45, dog_treat: 200, cat_toy: 40}
+```
+
 1. Given the following `states = {"CO" => "Colorado", "IA" => "Iowa", "OK" => "Oklahoma"}`, how would you access the value `"Iowa"`?
 
+```ruby
+puts states["IA"]
+```
+
 1. With the same hash above, how would we get all the keys?  How about all the values?
 
+```ruby
+#all keys
+puts states.keys
+
+#all values
+puts states.values
+```
+
 1. What is another example of when we might use a hash?  In your example, why is a hash better than an array?
 
+Another example of using a hash would be for seating arrangements for wedding guests. The key would be the guests' names and the values would be corresponding food allergies/preferences you want to remember about your guests. It might look like this:
+
+```ruby
+guests = {
+  'Naomi' => 'none'
+  'Ella' => 'shellfish',
+  'Al' => 'shellfish',
+  'Andrew' => 'vegan',
+  'Dan' => 'peanut',
+  'Henry' => 'none',
+  'Jill' => 'none',
+}
+```
+
+It would be harder to keep track of this information as an array because you would need to know which numerical index each person was at to access the information. That means you'd have to memorize the exact order the guests as listed in the array. If you looked up `guests[4]` in an array, it would return `Dan => peanut` because it's at the index position of 4. Let's say your partner rearranged the order of the guests.
+
+```ruby
+guests = {
+  'Ella' => 'shellfish',
+  'Al' => 'shellfish',
+  'Dan' => 'peanut',
+  'Andrew' => 'vegan',
+  'Naomi' => 'none'
+  'Henry' => 'none',
+  'Jill' => 'none',
+}
+```
+If you looked up `guests[4]` now, you would get `Naomi => none`.
+
+Using a hash would help you find the allergy that corresponds with `Dan` regardless of what order he's listed in the hash. You would find by simply checking for `guest['Dan']` and save Dan a trip to the ER by serving him peanut-free food.
+
 1. What questions do you still have about hashes?
+
+What are the use cases of the different syntaxes for hash? I see two in the examples of the exercises we've been doing:
```

Comment:
The main reason is whether or not we want to `symbolize` our information. This will come into play later down the road, however, it mostly ends up being a user preference

Check out for even more syntax options: https://launchschool.com/books/ruby/read/hashes#anoteonhashkeys

Side note: I LOVE that you consistently ask deeper questions - you want to know the whys, the hows, etc. Keep it up and don't lose this spirit! 
Sometimes code has multiple ways of accomplishing the same thing, sometimes there are minute differences between code snippets and learning why comes down to practice/research

Example of similar code with slight differences: Determining size can be accomplished in [multiple ways](https://batsov.com/articles/2014/02/17/the-elements-of-style-in-ruby-number-13-length-vs-size-vs-count/)

Second example: Printing can happen in [multiple ways](https://www.rubyguides.com/2018/10/puts-vs-print/)

Example of code [alias](https://medium.com/@janetmndz/understanding-rubys-inject-reduce-enumerable-methods-in-under-5-minutes-193e5cff06d0) (same exact code behavior, but difference names)